ISHMapboxDynamicFontObserver.m

```diff
@@ -0,0 +1,180 @@
+//
+//  ISHMapboxDynamicFontObserver.m
+//  MapboxDynamicFontStyle
+//
+//  Created by Felix Lamouroux on 18.08.17.
+//  Copyright © 2017 iosphere GmbH. All rights reserved.
+//
+
+#import <UIKit/UIKit.h>
+#import "ISHMapboxDynamicFontObserver.h"
+
+@interface ISHMapboxDynamicFontObserver ()
+@property (nonatomic) NSDictionary<NSString *, MGLStyleValue *> *baseTextSizeValues;
+@property (nonatomic) NSDictionary<NSString *, MGLStyleValue *> *baseIconSizeValues;
+@end
+
+@implementation ISHMapboxDynamicFontObserver
+
+#pragma mark - Setup
+
+- (instancetype)init {
+    self = [super init];
+
+    if (self) {
+        [[NSNotificationCenter defaultCenter] addObserver:self
+                                                 selector:@selector(UIContentSizeCategoryDidChangeNotification:)
+                                                     name:UIContentSizeCategoryDidChangeNotification
+                                                   object:nil];
+    }
+
+    return self;
+}
+
+- (instancetype)initWithStyle:(MGLStyle *)style {
+    self = [self init];
+    self.style = style;
+    return self;
+}
+
+- (void)setStyle:(MGLStyle *)style {
+    _style = style;
+
+    if (!style) {
+        self.baseTextSizeValues = nil;
+        return;
+    }
+
+    [self setupBaseValuesForStyle:style];
+    [self updateStyleForPreferredContentSizeCategory];
+}
+
+- (void)setupBaseValuesForStyle:(MGLStyle *)style {
+    NSMutableDictionary<NSString *, MGLStyleValue *> *baseTextSizeValues = [NSMutableDictionary new];
+    NSMutableDictionary<NSString *, MGLStyleValue *> *baseIconSizeValues = [NSMutableDictionary new];
+
+    for (MGLStyleLayer *layer in style.layers) {
```

Comment:
if we iterate over `MGLSymbolStyleLayer *layer` we still need the check below, but not the cast in l61